### PR TITLE
[Merged by Bors] - feat: a linear endomorphism that is a root of a squarefree polynomial is semisimple

### PR DESCRIPTION
**Pull request by ocfnash**
The main result is `Module.End.isSemisimple_of_squarefree_aeval_eq_zero`

---

- [x] depends on: #10086

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/DedekindDomain/Ideal.lean [hunk 1]
```diff
@@ -703,6 +703,16 @@ theorem Ideal.isPrime_iff_bot_or_prime {P : Ideal A} : IsPrime P ↔ P = ⊥ ∨
     hp.elim (fun h => h.symm ▸ Ideal.bot_prime) Ideal.isPrime_of_prime⟩
 #align ideal.is_prime_iff_bot_or_prime Ideal.isPrime_iff_bot_or_prime
 
+lemma Ideal.prime_span_singleton_of_prime {a : A} (h : Prime a) : Prime (Ideal.span {a}) := by
```
[the remark this answers is not in the document]
Comment (ocfnash) [reply Mathlib/RingTheory/DedekindDomain/Ideal.lean:706]: Great point!

File: Mathlib/LinearAlgebra/Semisimple.lean [hunk 2]
```diff
@@ -100,19 +104,39 @@ lemma eq_zero_of_isNilpotent_isSemisimple (hn : IsNilpotent f) (hs : f.IsSemisim
...
 
-lemma IsSemisimple_smul (t : K) (h : f'.IsSemisimple) :
-    (t • f').IsSemisimple := by
+lemma IsSemisimple_smul (t : K) (h : f.IsSemisimple) :
+    (t • f).IsSemisimple := by
   wlog ht : t ≠ 0; · simp [not_not.mp ht]
   rwa [IsSemisimple_smul_iff ht]
 
+open UniqueFactorizationMonoid in
+theorem isSemisimple_of_squarefree_aeval_eq_zero [FiniteDimensional K M]
```
[the remark this answers is not in the document]
Comment (ocfnash) [reply Mathlib/LinearAlgebra/Semisimple.lean:123]: Agreed.

**Comment by jcommelin** [comment]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Comment by ocfnash** [comment]
bors merge

**Pull request closed without merging**